auto_py_to_exe/web/js/interface.js

```diff
@@ -441,3 +441,38 @@ const setupLanguageSelection = () => {
     });
     languageSelectNode.value = currentLanguage;
 };
+
+// Toggle theme (triggered by clicking moon or sun)
+const _toggleTheme = () => {
+    const root = document.querySelector("body");
+    const onDarkThemeLabel = document.querySelector("#on-dark-theme-label");
+    const onLightThemeLabel = document.querySelector("#on-light-theme-label");
+    
+    if (root.classList.contains('dark-theme')) {
+        onLightThemeLabel.style.display = "inline";
+        onDarkThemeLabel.style.display = "none";
+    } else {
+        // dark
+        onLightThemeLabel.style.display = "none";
+        onDarkThemeLabel.style.display = "inline";
+    }
+
+    root.classList.toggle("dark-theme")
+}
+
+// Check if user's default color scheme is dark
+const setupTheme = () => {
+    const toggleButton = document.querySelector("span#theme");
+
+    toggleButton.addEventListener('click', _toggleTheme);
+
+    if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
+        const root = document.querySelector("body");
+        const onDarkThemeLabel = document.querySelector("#on-dark-theme-label");
+        const onLightThemeLabel = document.querySelector("#on-light-theme-label");
+
+        onLightThemeLabel.style.display = "none";
+        onDarkThemeLabel.style.display = "inline";
+        root.classList.add('dark-theme')
```

Comment:
Instead of duplicating logic, you should call `_toggleTheme` only if needed.

Also might pay to check some of this logic as when I opened the UI, it currently says "Theme: Light" and when I click it once, it still says this and then changes to dark move, consecutive clicks changes it.